src/containers/forms/PlanForm/helpers.ts

```diff
@@ -0,0 +1,339 @@
+import { FormikErrors } from 'formik';
+import { omit, pick } from 'lodash';
+import moment from 'moment';
+import { FormEvent } from 'react';
+import * as Yup from 'yup';
+import { DATE_FORMAT, DEFAULT_ACTIVITY_DURATION_DAYS } from '../../../configs/env';
+import {
+  actionReasons,
+  ActionReasonType,
+  FIClassifications,
+  FIReasons,
+  FIReasonType,
+  FIStatusType,
+  goalPriorities,
+  GoalPriorityType,
+  PlanAction,
+  PlanActionCodes,
+  PlanActionCodesType,
+  planActivities,
+  PlanActivity,
+  PlanDefinition,
+  PlanGoal,
+  PlanGoalDetail,
+  PlanGoaldetailQuantity,
+  PlanGoalTarget,
+  UseContext,
+} from '../../../configs/settings';
+import { DATE, IRS_TITLE, IS, NAME, REQUIRED } from '../../../constants';
+import { InterventionType, PlanStatus } from '../../../store/ducks/plans';
+
+/** separate FI and IRS activities */
+export const FIActivities = omit(planActivities, ['IRS']);
+export const IRSActivities = pick(planActivities, ['IRS']);
+
+/** Array of FI Statuses */
+export const fiStatusCodes = Object.values(FIClassifications).map(e => e.code as FIStatusType);
+
+/** Yup validation schema for PlanForm */
+export const PlanSchema = Yup.object().shape({
+  activities: Yup.array().of(
+    Yup.object().shape({
+      actionCode: Yup.string().oneOf(PlanActionCodes.map(e => e)),
+      actionDescription: Yup.string().required(REQUIRED),
+      actionReason: Yup.string()
+        .oneOf(Object.values(actionReasons))
+        .required(REQUIRED),
+      actionTitle: Yup.string().required(REQUIRED),
+      goalDescription: Yup.string().required(REQUIRED),
+      goalDue: Yup.date().required(REQUIRED),
+      goalPriority: Yup.string()
+        .oneOf(Object.values(goalPriorities))
+        .required(REQUIRED),
+      goalValue: Yup.number()
+        .min(1)
+        .required(REQUIRED),
+      timingPeriodEnd: Yup.date().required(REQUIRED),
+      timingPeriodStart: Yup.date().required(REQUIRED),
+    })
+  ),
+  caseNum: Yup.string(),
+  date: Yup.string().required(`${DATE} ${IS} ${REQUIRED}`),
+  end: Yup.date().required(REQUIRED),
+  fiReason: Yup.string().oneOf(FIReasons.map(e => e)),
+  fiStatus: Yup.string().oneOf(fiStatusCodes),
```

Comment:
Little nit-picky, feels like `fiReason` and `fiStatus` should be mapped the same way (here or above) for uniformity.

Reply:
Hmmm, could you explain further?

Reply:
Just that `FIReasons` is mapping inline while `fiStatusCodes` was mapped previously.

Reply:
Created an issue: https://github.com/onaio/reveal-frontend/issues/253